### PR TITLE
Transformer for CLC100

### PR DESCRIPTION
**Pull request by ulziibay**

### CODE REVIEW

File: flake8_click.py [hunk 1]
```diff
@@ -1,44 +1,75 @@
...
-        arg_names: List[str] = [arg.arg for arg in f.args.args]
+    def get_func_arguments(self, f: Union[ast.FunctionDef, cst.FunctionDef])-> List[str]:
+        arg_names: List[str] = []
+        if isinstance(f, ast.FunctionDef):
+            arg_names = [arg.arg for arg in f.args.args]
+        else:
+            arg_names = [param.name for param in f.params.params]
         return arg_names
 
+    def get_option_param_name(self, d: Union[ast.Call, cst.Call]) -> str:
```
Comment (nbrahms) [inline flake8_click.py]: Note that there's a bug here when `d.args` is empty.

I fixed it in my PR, but you'll need to resolve conflicts.

File: flake8_click.py [hunk 2]
```diff
@@ -156,5 +172,95 @@ def run(self):
                     "ClickOptionFunctionArgumentChecker",
                 )
 
-    def _message_for(self, func_def: ast.FunctionDef, options: List[str]):
+    def _message_for(self, func_def: Union[ast.FunctionDef, cst.FunctionDef], options: List[str]):
         return f"CLC100: function `{self.get_name_func(func_def)}` missing parameter `{','.join(options)}` for `@click.option`{'-s' if len(options) > 0 else ''}"
+
+
+@attr.s(auto_attribs=True)
+class ClickCommandArgumentAddTransformer(cst.CSTTransformer):
```
Comment (nbrahms) [inline flake8_click.py]: It seems like all the code here is duplicated with `ClickCommandVisitor`. Can you make a mixin to handle this functionality?

File: flake8_click.py [hunk 3]
```diff
@@ -156,5 +172,95 @@ def run(self):
...
+            if not isinstance(decorator, cst.Call):
+                continue
+            if self.is_click_command(decorator):
+                use_cli_command = True
+            elif self.is_click_option(decorator):
+                if not use_cli_command:
+                    continue
+                param_name = self.get_option_param_name(decorator)
+                if not self.param_in_function_def(f, param_name):
+                    option_param_names.append(param_name)
```
Comment (nbrahms) [inline flake8_click.py]: Should use iterator work here.

File: flake8_click.py (hunk 2 above)
Comment (ulziibay) [reply flake8_click.py]: Yep. This is demoware. Will rewrite most of it!

File: flake8_click.py (hunk 3 above)
Comment (ulziibay) [reply flake8_click.py]: Will do

**Pull request closed without merging**